### PR TITLE
Feat: Add collateral transfer to treasury

### PR DESCRIPTION
**Pull request by Zitzak**
## What has been done?
- Rebase locally from the Inverter feature branch
- Add treasury address to state, as well as setters for it
- overridden `_handleCollateralTokensBeforeBuy` to transfer collateral out of the contract during buy operations

**Please Note:** The branch `dev-1` first has to be rebased to resolve the merge conflicts

### CODE REVIEW

File: src/modules/fundingManager/oracle/FM_PC_ExternalPrice_Redeeming_v1.sol [hunk 1]
```diff
@@ -273,6 +279,11 @@ contract FM_PC_ExternalPrice_Redeeming_v1 is
         return _orderId;
     }
 
+    /// @inheritdoc IFM_PC_ExternalPrice_Redeeming_v1
+    function getProjctTreasury() external view returns (address) {
```
Comment (vendrell46) [inline src/modules/fundingManager/oracle/FM_PC_ExternalPrice_Redeeming_v1.sol]: typo `getProjectTreasury`*

File: src/modules/fundingManager/oracle/interfaces/IFM_PC_ExternalPrice_Redeeming_v1.sol [hunk 2]
```diff
@@ -76,7 +76,10 @@ interface IFM_PC_ExternalPrice_Redeeming_v1 is
     /// @notice	Thrown when a redemption queue execution fails
     error Module__FM_PC_ExternalPrice_Redeeming_QueueExecutionFailed();
 
-    // -------------------------------------------------------------------------
+    /// @notice Thrown when the project treasury address is invalid
+    error Module__FM_PC_ExternalPrice_Redeeming_InvalidProjectTreasury();
+
+    //--------------------------------------------------------------------------
```
Comment (vendrell46) [inline src/modules/fundingManager/oracle/interfaces/IFM_PC_ExternalPrice_Redeeming_v1.sol]: a space between `//` and the `---` is missing

File: src/modules/fundingManager/oracle/FM_PC_ExternalPrice_Redeeming_v1.sol (hunk 1 above)
Comment (Zitzak) [reply src/modules/fundingManager/oracle/FM_PC_ExternalPrice_Redeeming_v1.sol]: Done

File: src/modules/fundingManager/oracle/interfaces/IFM_PC_ExternalPrice_Redeeming_v1.sol [hunk 3]
```diff
@@ -127,7 +130,11 @@ interface IFM_PC_ExternalPrice_Redeeming_v1 is
     /// @return	orderId_ The current order ID
     function getOrderId() external view returns (uint orderId_);
 
-    // -------------------------------------------------------------------------
+    /// @notice Gets the project treasury address
+    /// @return address_ The address of the project treasury
+    function getProjctTreasury() external view returns (address);
```
Comment (vendrell46) [inline src/modules/fundingManager/oracle/interfaces/IFM_PC_ExternalPrice_Redeeming_v1.sol]: typo `getProjectTreasury`*

**Review by vendrell46 — APPROVED** [review]

File: src/modules/fundingManager/oracle/interfaces/IFM_PC_ExternalPrice_Redeeming_v1.sol (hunk 2 above)
Comment (Zitzak) [reply src/modules/fundingManager/oracle/interfaces/IFM_PC_ExternalPrice_Redeeming_v1.sol]: Done

**Pull request merged**